extension/zpagesextension/zpagesextension.go

```diff
@@ -69,12 +91,27 @@ func (zpe *zpagesExtension) Shutdown(context.Context) error {
 	if zpe.stopCh != nil {
 		<-zpe.stopCh
 	}
+
+	// Try to unregister the zpages span processor with the UnregisterSpanProcessor method.
+	// https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#TracerProvider.UnregisterSpanProcessor.
+	// Use an anonymous interface to support non-SDK trace providers.
```

Comment:
nit: I think is a good reason, very unlikely since the processor is a SDK concept :)

Reply:
If people want to wrap the official SDK with something of their own we can support them in this way, while we wouldn't be able to otherwise, I think that's a good enough reason to keep it with an interface